lib/rust/profiling/macros/src/lib.rs

```diff
@@ -0,0 +1,109 @@
+//! Support crate that allows to define loggers based on a configuration file.
+#![warn(missing_debug_implementations)]
+#![warn(missing_docs)]
+#![warn(trivial_casts)]
+#![warn(trivial_numeric_casts)]
+#![warn(unsafe_code)]
+#![warn(unused_import_braces)]
+
+use proc_macro2::Ident;
+use proc_macro2::TokenStream;
+use proc_macro2::TokenTree;
+use quote::quote;
+
+/// Config file that contains a list of the loggers that should exist.
+const PROFILERS: &str =
+    include_str!("../../../../../app/ide-desktop/lib/profiling/src/profilers.json");
+
+/// Calls the given macro with each defined profiler and creates a method that checks whether
+/// a given profiler is activated via the crates feature flags.
+/// The macro that is expected to be passed is `define_logger` from the profiling crate. The
+/// call is built to satisfy the input requirements of this macro. The `define_logger` macro
+/// needs to be defined in the profiling crate as it exposes API related structs that also need
+/// to be part of the public API there and to avoid circular dependencies.
+#[proc_macro]
+pub fn with_all_profiling_levels(item: proc_macro::TokenStream) -> proc_macro::TokenStream {
+    let input: TokenStream = item.into();
+    let input_macro: Ident = {
+        match input.into_iter().next().unwrap() {
+            TokenTree::Ident(i) => i,
+            _ => panic!("`with_all_profiling_levels` requires an identifier as input."),
+        }
+    };
+    let profilers: Vec<String> = serde_json::from_str(PROFILERS).unwrap();
+
+    let mut out_tokens = proc_macro::TokenStream::default();
+    out_tokens.extend(make_profilers(input_macro, &profilers).into_iter());
+    out_tokens.extend(make_profiling_level_is_active_function(&profilers).into_iter());
+
+    out_tokens
+}
```

Comment:
Please do this as we agreed on Discord. With proc macro, just generate a macro_rules macro named `with_all_profiling_levels` which will behave like this: if I call `with_all_profiling_levels!(foo)` it will expand to `foo!([task start_task end_task] [section start_section end_section] [...])`. This way, the proc macro file would be MUCH smaller in size and all the other things could be implemented just as macro_rules. If you want to change the design we agreed upon, lets have a call and discuss it first, please.